server/middleware.ts

```diff
@@ -104,7 +104,7 @@ export function isAdmin(request: express.Request, response: express.Response, ne
 // For API endpoints
 export function authenticateWithReject(request: express.Request, response: express.Response, next: express.NextFunction) {
 	response.setHeader("Cache-Control", "no-cache");
-	if (!request.isAuthenticated()) {
+	if (!request.isAuthenticated() || !request.user || !request.user.verifiedEmail || !request.user.accountConfirmed) {
```

Comment:
Is it possible to have passport check this so we can simplify the logic to just `request.isAuthenticated()` everywhere else?